### ISSUE TITLE
Linter should warn if 'let'/'const' and 'var' are used in the same file.

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#1881]
The following should generate a linter warning:
```js
let foo = 1;
var bar = 2;
```

**Comment by Dominator008** [comment]
We might as well go further and have a lint check that warns on any `var

**Comment by mashedcode** [comment]
@Dominator008 my guess would be that people would do `s/var/let/` just to get rid of the errors and that would be just as terrible because there's a reason there's `const`. So please provide an option to turn that warning off if you do so.

**Comment by ChadKillingsworth** [comment]
There are definite cases for `var`. Specifically we only support typdefs as vars because we expect hoisting.
